### PR TITLE
Fix pytest_report_teststatus hook

### PR DESCRIPTION
**Pull request by blueyed**
This calls `pytest_report_teststatus` through `self.config.hook`, and
does not return anything for "setup" and "teardown" reports in our own
`pytest_report_teststatus`, so it falls back to the runner's
implementation, which handles "setup" and "teardown".

Fixes https://github.com/Frozenball/pytest-sugar/issues/94.

### CODE REVIEW

**Comment by Frozenball** [comment]
I think this has been fixed now when I modified setup / teardown errors to be treated as errors.

File: pytest_sugar.py [hunk 1]
```diff
@@ -362,9 +365,6 @@ def pytest_runtest_logreport(self, report):
             elif self.reached_last_column_for_test_status():
                 self.begin_new_line(report, print_filename=False)
 
-            rep = report
-            res = pytest_report_teststatus(report=report)
-            cat, letter, word = res
```
Comment (blueyed) [inline pytest_sugar.py]: @Frozenball 
This seems to still be not necessary.
Should I create a new PR for this?!

**Comment by blueyed** [comment]
Thanks!
For reference: https://github.com/Frozenball/pytest-sugar/commit/93bf75d387151c5821df6e60a938ee42d4902609.

However, there is still code here that makes sense (see my comment above).

File: pytest_sugar.py (hunk 1 above)
Comment (Frozenball) [inline pytest_sugar.py]: Sure!

Comment (blueyed) [inline pytest_sugar.py]: https://github.com/Frozenball/pytest-sugar/pull/110

**Pull request closed without merging**